opentelemetry-sdk/benches/metrics_counter.rs

```diff
@@ -57,62 +57,72 @@ fn criterion_benchmark(c: &mut Criterion) {
 fn counter_add_sorted(c: &mut Criterion) {
     let counter = create_counter("Counter_Add_Sorted");
     c.bench_function("Counter_Add_Sorted", |b| {
-        b.iter(|| {
-            // 4*4*10*10 = 1600 time series.
-            let rands = CURRENT_RNG.with(|rng| {
-                let mut rng = rng.borrow_mut();
-                [
-                    rng.gen_range(0..4),
-                    rng.gen_range(0..4),
-                    rng.gen_range(0..10),
-                    rng.gen_range(0..10),
-                ]
-            });
-            let index_first_attribute = rands[0];
-            let index_second_attribute = rands[1];
-            let index_third_attribute = rands[2];
-            let index_fourth_attribute = rands[3];
-            counter.add(
-                1,
-                &[
-                    KeyValue::new("attribute1", ATTRIBUTE_VALUES[index_first_attribute]),
-                    KeyValue::new("attribute2", ATTRIBUTE_VALUES[index_second_attribute]),
-                    KeyValue::new("attribute3", ATTRIBUTE_VALUES[index_third_attribute]),
-                    KeyValue::new("attribute4", ATTRIBUTE_VALUES[index_fourth_attribute]),
-                ],
-            );
-        });
+        b.iter_batched(
+            || {
+                // 4*4*10*10 = 1600 time series.
+                CURRENT_RNG.with(|rng| {
+                    let mut rng = rng.borrow_mut();
+                    [
+                        rng.gen_range(0..4),
+                        rng.gen_range(0..4),
+                        rng.gen_range(0..10),
+                        rng.gen_range(0..10),
+                    ]
+                })
+            },
+            |rands| {
+                let index_first_attribute = rands[0];
+                let index_second_attribute = rands[1];
+                let index_third_attribute = rands[2];
+                let index_fourth_attribute = rands[3];
+                counter.add(
+                    1,
+                    &[
+                        KeyValue::new("attribute1", ATTRIBUTE_VALUES[index_first_attribute]),
+                        KeyValue::new("attribute2", ATTRIBUTE_VALUES[index_second_attribute]),
+                        KeyValue::new("attribute3", ATTRIBUTE_VALUES[index_third_attribute]),
+                        KeyValue::new("attribute4", ATTRIBUTE_VALUES[index_fourth_attribute]),
+                    ],
+                );
+            },
+            BatchSize::LargeInput,
```

Comment:
Does it need to be `BatchSize::LargeInput`? Could we use `BatchSize::SmallInput` instead as it has a lower measurement overhead?

Reply:
The only reason of using large batch I can think here is if the setup overhead is significant, and the large batch size relatively reduces the impact. But otherwise, small batch size is good enough.

Reply:
From the [docs](https://docs.rs/criterion/latest/criterion/enum.BatchSize.html):

> Generally speaking, almost all benchmarks should use `SmallInput`. If the input or the result of the benchmark routine is large enough that `SmallInput` causes out-of-memory errors, `LargeInput` can be used to reduce memory usage at the cost of increasing the measurement overhead.

I would prefer using `SmallInput` whenever possible.

Reply:
Good catch! Fixed. PTAL thanks.

Reply:
@zarvd It would be great if you could also confirm that running these benchmarks with `SmallInput` did not run into any issues.

Reply:
It’s working fine on my Mac and the Ubuntu VM. I’ve also updated the PR description with the results.